core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -156,6 +156,16 @@ public List<HistoryEntry> history() {
     return table().history();
   }
 
+  @Override
+  public SnapshotRef ref(String ref) {
```

Comment:
feels a bit awkward for metadata table to have refs... but given the fact that we are just mapping this to methods of the source `table()` for all the other methods, this seems to be the only way to do it